server/UserRegistration.java

```diff
@@ -0,0 +1,467 @@
+package newbank.server;
+
+import com.sun.tools.javac.Main;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Scanner;
+import java.util.*;
+import java.sql.SQLException;   // If have an error thrown from execution.
+
+public class UserRegistration extends Main {
+
+
+    public static UserRegistration newUser() throws SQLException {
+
+        Scanner sc = new Scanner(System.in);
+        // set all methods to return values - created conversions for sortcode
+        // and account number to make it easier to push through to db.
+
+
+        newUserName nn = new newUserName();
+        String userFullname = nn.GetNameFromUser();
+
+
+        DateChecks dc = new DateChecks();
+        String userDoB = dc.GetDateFromUser();
+
+
+        UserId ui = new UserId();
+        String userUsername = ui.getNewUserId();
+
+
+        newUserAccountNumber acm = new newUserAccountNumber();
+        String userAccountNumber = acm.getNewUserAccountNumber();
+
+
+        newUserSortcode nusort = new newUserSortcode();
+        String userSortCode = nusort.getNewUserSortCode();
+
+
+        NewUserSecurity nus = new NewUserSecurity();
+
+        String userNewPass = nus.inputNewPassword();
+
+
+        NewUserBalance nub = new NewUserBalance();
+
+        Integer userNewBal = Integer.valueOf(nub.balance());
+
+
+        newUserSecretQuestion nsq = new newUserSecretQuestion();
+
+        String userNewSecretQuestion = nsq.secretQuestion();
+
+
+        checkNotARobot cnr = new checkNotARobot();
+
+        boolean NewUserNotARobot = cnr.checkHuman();
+
+        // All already scanned by scanner then passed as values to 'String addNewUser'
+        // The values are then passed to the db.
+
+
+        Connection conn = null;
+
+        try {
+            String url = "jdbc:sqlite:/Users/pmcdowell/Documents/nb/database/bank.sq3";
+
+            conn = DriverManager.getConnection(url);
+            // Will allow for the database connection.
+
+            String addNewUser = "INSERT INTO useroverview(name_user, dob_user, username_user, accountNumber_user, sortcode_user, password_user, balance_user, secretAns_user) VALUES (?, ?, ?, ?, ?, ?, ?, ?)";
```

Comment:
This comment is in regard to the database/table design. I'd name the table `users` since it's about users of the system, and then I'd remove the `_user` suffix, as it's redundant, since the table name is `users`

Also, more of a nitpick, but it's a convention to not use camelCase for column names, so you'd for example do `account_number` rather than `accountNumber`, but it's not critical to change it now